### PR TITLE
tests: do not test dependent formulae with :all bottles on Linux

### PR DESCRIPTION
**Pull request by iMichka**
See:
https://github.com/Homebrew/homebrew-core/pull/80242

This prevents endless builds, and tests on formulae that are
not yet ready for Linux

### CODE REVIEW

**Comment by iMichka** [comment]
I think you would have hit the same issue with ARM bottling, but as `all:` bottles were introduced later, this was not an issue.

File: lib/tests/formulae.rb [hunk 1]
```diff
@@ -376,7 +376,17 @@ def setup_formulae_deps_instances(formula, formula_name, args:)
...
         @testable_dependents = @source_dependents.select(&:test_defined?)
         @bottled_dependents = with_env(HOMEBREW_SKIP_OR_LATER_BOTTLES: "1") do
-          dependents.select(&:bottled?)
+          if OS.linux? &&
+             tap.present? &&
+             tap.full_name == "Homebrew/homebrew-core" &&
+             ENV["HOMEBREW_SKIP_UNBOTTLED_LINUX_TESTS"]
+            # :all bottles are considered as Linux bottles, but as we did not bottle
+            # everything (yet) in homebrew-core, we do not want to test formulae
+            # with :all bottles for the time being.
```
Comment (carlocab) [inline lib/tests/formulae.rb:385]: Do we want to add a `TODO` to the comment?

**Review by carlocab — APPROVED** [review]

File: lib/tests/formulae.rb (hunk 1 above)
Comment (iMichka) [reply lib/tests/formulae.rb:385]: We will remove HOMEBREW_SKIP_UNBOTTLED_LINUX_TESTS eventually so this block of code will be cleaned up.

**Comment by dawidd6** [comment]
Imma test this in:
https://github.com/Homebrew/homebrew-core/pull/80354/

**Comment by iMichka** [comment]
Thanks. I retriggered the ncurses build, and will dispatch a job for `alsa-lib`, which is the formula which was impacted too.

**Comment by dawidd6** [comment]
Nothing really changed:
https://github.com/Homebrew/homebrew-core/runs/2959377767?check_suite_focus=true

Huh

**Comment by iMichka** [comment]
Weird. I tested the change locally in a docker container and it worked.

I think I ran something like this to test it:

```
docker run -it ghcr.io/homebrew/ubuntu16.04:master
git remote set-url origin https://github.com/Homebrew/homebrew-core
brew update
export HOMEBREW_FORCE_HOMEBREW_ON_LINUX=1
export HOMEBREW_SKIP_UNBOTTLED_LINUX_TESTS=1
brew test-bot --only-formulae alsa-lib
```

**Comment by iMichka** [comment]
@dawidd6 for typescript we need to add the same check somewhere else: https://github.com/Homebrew/homebrew-test-bot/blob/master/lib/tests/formulae.rb#L618-L627

I disabled `all` builds for dependents, but not for the main tested formula.

I am still trying to understand what is going on with the other builds on CI, as my fix seems to have no effect.

**Pull request merged**